Settings/BusinessLayer/Services/SettingsService.cs

```diff
@@ -20,25 +21,41 @@ public SettingsService(SettingsRepository settingsRepository, IMapper mapper)
         }
         public async Task AddSettingAsync(SettingDto settingDto,int userId, CancellationToken token)
         {
+            if (await _settingsRepository.IsEntityExistsAsync(userId.ToString(), token))
+            {
+                throw new ApiException("Invalid operation", ExceptionStatus.BadRequest);
+            }
             var setting = _mapper.Map<SettingsInfo>(settingDto);
             setting.Id = userId.ToString();
             await _settingsRepository.CreateAsync(setting,token);
         }
 
         public async Task DeleteSettingAsync(int id, CancellationToken token)
         {
+            if (!(await _settingsRepository.IsEntityExistsAsync(id.ToString(),token)))
+            {
+                throw new ApiException("Invalid operation", ExceptionStatus.BadRequest);
```

Comment:
for the future, create more infromative messages and create class with constants and these constants will contain the exception messages